js/ui/codingManager.js

```diff
@@ -81,11 +81,19 @@ const CodingManager = new Lang.Class({
     },
 
     addBuilderWindow: function(actor) {
+        let tracker = Shell.WindowTracker.get_default();
+        let correspondingApp = tracker.get_window_app(actor.meta_window);
+        let isSpeedwagonForBuilder = (
+            Shell.WindowTracker.is_speedwagon_window(actor.meta_window) &&
+            (correspondingApp ? correspondingApp.get_id() === 'org.gnome.Builder.desktop' : false)
```

Comment:
Ah - for some reason I had the impression that JS didn't short-circuit, but it looks like I am wrong: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators . Fixed